tool/src/shared.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import "dart:async";
+import "dart:io";
+import "dart:convert";
+
+// Shared tools used by other libraries.
+
+/// Quick and dirty URI loader.
+///
+/// Stashes copy in specified file, or in file in tmp directory.
+Future<String> fetch(String location,
+    {File? targetFile, bool forceLoad = false}) async {
+  if (targetFile == null) {
+    String safeLocation = location.replaceAll(RegExp(r"[^\w]+"), "-");
+    targetFile = File(path(Directory.systemTemp.path, safeLocation));
+  }
+  if (!forceLoad && targetFile.existsSync()) {
+    return targetFile.readAsString();
+  }
+  Uri uri = Uri.parse(location);
```

Comment:
We should consider turning on omit_local_variable_types and cleaning this up. I could do that in a followup.

Reply:
Done.